Code/Mantid/Framework/API/src/Projection.cpp

```diff
@@ -38,6 +40,64 @@ Projection::Projection(const V3D &u, const V3D &v, const V3D &w) {
   }
 }
 
+Projection::Projection(ITableWorkspace_const_sptr ws) {
```

Comment:
I don't insist but consider these points:

1. Workspaces usually come from the ADS which checks for workspace existence.
2. If you have a workspace outside ADS you'll have to wrap it into a shared pointer to pass to Projection.
3. Calling Projection(*workspace) doesn't require lots of manual work.
4. A result of passing an empty shared pointer is an exception which you'll have to catch, the same as with *workspace.
5. "nicer" is not an argument.